### PR TITLE
Few improvements to Cloud NAT

### PR DESCRIPTION
**Pull request by mborsz**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Few tiny improvements to how Cloud NAT routers are created:
* add a way to override default 64 ports per vm
* make create-cloud-nat-router idempotent (useful when two clusters are created in the same network)
* modify check-resources to not detect cloud nat as it should be interpreted as network-wide object, not cluster-wide (again, this prevents creating kubemark masters and gce masters in the same network)

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```
/assign @wojtek-t

### CODE REVIEW

File: cluster/gce/util.sh [hunk 1]
```diff
@@ -3464,13 +3469,6 @@ function check-resources() {
     return 1
   fi
 
-  if [[ ${GCE_PRIVATE_CLUSTER:-} == "true" ]]; then
```
Comment (wojtek-t) [inline cluster/gce/util.sh]: Why did you remove this one?

Comment (wojtek-t) [reply cluster/gce/util.sh]: It was used to detect whether we didn't leak cloud nat.

Comment (mborsz) [reply cluster/gce/util.sh]: check-resources is used only here:

https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/util.sh#L2323

It checks if there is no other cluster running in the network with the same name and asks if kube-down should be called.

It's a bug that we check there for existence of cloud-nat, which is created per network and not per cluster.

If we will leave this here, any attempt to create a second private cluster in the same network (e.g. in kubemark) will ask for deleting the previous one which is incorrect.

**Comment by mborsz** [comment]
Rebased

**Comment by mborsz** [comment]
pull-kubernetes-kubemark-e2e-gce-big seems to be deterministically  failing.
/hold

**Comment by mborsz** [comment]
After https://github.com/kubernetes/test-infra/pull/14928 I expect this test to start passing.
/hold cancel
